ifrs17/Utils/ActivityLog.ipynb

```diff
@@ -354,6 +354,160 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record ImportBuilderWriter(ImportOptionsBuilder Builder, ",
+                "\n                                ISessionVariable Session, ",
+                "\n                                IDataSource DataSource, ",
+                "\n                                IDataSetImportVariable ImportVariable)",
+                "\n{",
+                "\n    public async Task<ActivityLog> ExecuteAsync()",
+                "\n    {",
+                "\n        var log = await Builder.ExecuteAsync();",
+                "\n        var options = Builder.GetImportOptions();",
+                "\n        var activity = new ImportExportActivity(log, Session);",
+                "\n        bool importSucceeded = true;",
+                "\n        switch(options)",
+                "\n        {",
+                "\n            case FileImportOptions fio:",
+                "\n                var importFile = new ImportFile(Guid.NewGuid());",
+                "\n                try",
+                "\n                {",
+                "\n                    importFile = await (new ImportFile(fio, ImportVariable, Session)).InitializeImportDataAsync() as ImportFile;",
+                "\n                    activity = activity with {SourceId = importFile.Id};",
+                "\n                }",
+                "\n                catch (Exception)",
+                "\n                {",
+                "\n                    importSucceeded = false;",
+                "\n                    activity = activity with {SourceId = null};",
+                "\n                }",
+                "\n                activity = activity with {Category = \"Import from File\"};",
+                "\n                if (importSucceeded) await DataSource.UpdateAsync<ImportFile>(importFile.RepeatOnce());",
+                "\n                break;",
```

Comment:
Can't you simplify this:
1. Add the "await DataSource.UpdateAsync<ImportFile>(importFile.RepeatOnce());"," inside the try
2. Remove importSucceeded,
3. Add "activity = activity with {Category = \"Import from File\"};"," inside the try and catch --> you are creating activity twice (once to set SourceId, once to set Category).
4. Declare importFile inside the try
5. Extract the try/catch as method of type <T> and pass the parameters to avoid the repetition of this code.

I have not thought it through, but I guess this should work and make the code leaner

Reply:
the pattern matching can then be used instead of the switch case.

Reply:
I suggest to close this PR as is to give clients the possibility to adapt to the new api. 
Improvements in the code can be done in a separate pr. 
The comments are captured in #167